launchprocess.sh

```diff
@@ -0,0 +1,110 @@
+#!/bin/bash
+#script to launch a specified TorQ process (standard or custom)
+#For now the exit codes are: 
+#1:procname/proctype not provided
+#2:process failed to start
+#3:process had already been running
+
+#for each cmd line argument, check if it is in the default set
+cnt=0
+IFS='-'
+read -ra arg_arr <<< "$@"
+if [ "${arg_arr[1]:0:8}" == "procname" ] && [ "${arg_arr[2]:0:8}" == "proctype" ]; then
+    for arg in "${arg_arr[@]}"
+    do
+        if [ "$cnt" -gt 0 ]; then
+        IFS=' '
+        read -ra sub_arg_arr <<< "$arg"
+        case ${sub_arg_arr[0]} in 
+	    procname )
+		procname=${sub_arg_arr[1]}
+		;;
+	    proctype )
+		proctype=${sub_arg_arr[1]}
+		;;
+	    U )
+		U_arg=${sub_arg_arr[1]}
+		;;
+	    localtime )
+	        localtime_arg=${sub_arg_arr[1]}
+		;;
+	    g )
+		g_arg=${sub_arg_arr[1]}
+		;;
+	    T )
+		T_arg=${sub_arg_arr[1]}
+		;;
+	    w )
+		w_arg=${sub_arg_arr[1]}
+		;;
+	    qcmd )
+		qcmd_arg=${sub_arg_arr[1]}
+		;;
+	    * )
+		other_args=${sub_arg_arr[1]}
+		;;
+        esac
+        fi
+        cnt=$((cnt+1))	
+    done
+else 
+    echo "Error: First two parameters must be procname and proctype"
+    exit 1
+fi
+
+#set default values
+U_arg=${U_arg:=0}
```

Comment:
U is meant to be path to file with usernames & passwords, so I don't think a default value of 0 is appropriate here. Possibly we don't default this here & instead default it in the wrapper q function, where it'll be easier to get the location of the calling process' -U file to use as a default?

Reply:
Ok sounds good.